applications/ShallowWaterApplication/custom_utilities/estimate_dt_utility.cpp

```diff
@@ -0,0 +1,135 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "estimate_dt_utility.h"
+#include "utilities/openmp_utils.h"
+#include "processes/find_nodal_h_process.h"
+#include "shallow_water_application_variables.h"
+
+
+namespace Kratos
+{
+
+EstimateDtShallow::EstimateDtShallow(
+    ModelPart& rThisModelPart,
+    Parameters& rThisParameters
+) : mrModelPart(rThisModelPart)
+{
+
+    Parameters default_parameters(R"({
+        "automatic_time_step"   : true,
+        "time_step"             : 1.0,
+        "courant_number"        : 1.0,
+        "consider_froude"       : true,
+        "compute_nodal_h"       : false,
+        "minimum_delta_time"    : 1e-4,
+        "maximum_delta_time"    : 1e+6
+    })");
+
+    rThisParameters.ValidateAndAssignDefaults(default_parameters);
+
+    mEstimateDt = rThisParameters["automatic_time_step"].GetBool();
+    mConstantDt = rThisParameters["time_step"].GetDouble();
+    mCourant = rThisParameters["courant_number"].GetDouble();
+    mConsiderFroude = rThisParameters["consider_froude"].GetBool();
+    mComputeNodalH = rThisParameters["compute_nodal_h"].GetBool();
+    mMinDt = rThisParameters["minimum_delta_time"].GetDouble();
+    mMaxDt = rThisParameters["maximum_delta_time"].GetDouble();
+    mEpsilon = std::numeric_limits<double>::epsilon();
+
+    if (mComputeNodalH)
```

Comment:
```suggestion
    if (rThisParameters["compute_nodal_h"].GetBool()) {
```
same reason, this member is not used anywhere else

Reply:
i've removed since i'm calling the find_nodal_h_process from the python script